plugin/path_role_set.go

```diff
@@ -318,7 +318,7 @@ func (b *backend) pathRoleSetCreateUpdate(ctx context.Context, req *logical.Requ
 		warnings = append(warnings, updateWarns...)
 	}
 	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+		return logical.ErrorResponse(err.Error()), err
```

Comment:
How is `err` in the second parameter used here (I can't remember how the framework deals with it)? I noticed that a few lines up we're returning an error response with `nil`.

Reply:
The CLI handles the logical response being an error, but in writing a test I had to inspect the response for the error, which felt weird instead of just checking of error was nil. See https://github.com/hashicorp/vault-plugin-secrets-gcp/pull/136/files#diff-6307edd822eb0e21be96d45f609fcf26677d56545380eb9392d625b6ac6b3cb7R151-R160

Reply:
I'm not sure that we're swallowing an error in this case. Returning `logical.ErrorResponse` and a nil error is common for user errors.

Last I checked, the framework would only use the `logical.ErrorResponse` when it was returned along with a non-nil error. This ends up resulting in a 400 Bad Request. Returning the error along with a `logical.ErrorResponse` didn't make a difference from the perspective of an HTTP client. 

From the perspective of the test, you could instead:
```
resp, _ = testRoleSetCreateRaw()
if !resp.IsError() {
    t.Fatal("expected error, got none")
}
```

Reply:
@austingebauer  is right -- we can either `return logical.ErrorResponse, nil` or `return nil, err`, but don't have to do both. On tests, it does make error-checking a bit more tedious since you'd have to do something like ` err != nil || (resp != nil && resp.IsError()` but in either case they it result in a non-200 response.

Reply:
Done.